hack/create-dev-cluster.sh

```diff
@@ -38,6 +38,11 @@ export AZURE_VNET_NAME=${CLUSTER_NAME}-vnet
 export AZURE_LOCATION="${AZURE_LOCATION:-southcentralus}"
 export AZURE_RESOURCE_GROUP=${CLUSTER_NAME}
 
+AZURE_SUBSCRIPTION_ID="${AZURE_SUBSCRIPTION_ID:=}"
+AZURE_TENANT_ID="${AZURE_TENANT_ID:=}"
+AZURE_CLIENT_ID="${AZURE_CLIENT_ID:=}"
+AZURE_CLIENT_SECRET="${AZURE_CLIENT_SECRET:=}"
```

Comment:
Why are these now needed for workload identity?

Reply:
They aren't, these initialization statements are required in order to say "you don't have to declare these variables ahead of time". Without initializing default values the script will complain about unbound variables if they aren't set in the runtime environmental context.